### ISSUE TITLE
item.to.getAsync removes unresolved mail addresses

### ISSUE THREAD

**Issue by rgmills** [OfficeDev/office-js#738]
Moving to a new issue from https://github.com/OfficeDev/office-js/issues/486#issuecomment-534160039

In the case of calling `Office.context.mailbox.item.to.getAsync(callback);` when there are only contacts who are not resolved, Outlook is removing those contacts from the To recipient list.

I've checked in the Outlook Web client and it does not remove the unresolved smtp email address.

## Expected Behavior
Calling Office.context.mailbox.item.to.getAsync() should return resolved recipients without side effects

## Current Behavior
Unresolved recipients are removed

## Steps to Reproduce, or Live Example
- Open new email message
- Manually type in an smtp email address, hit esc to switch focus
- Invoke Office.context.mailbox.item.to.getAsync(callback); in an add-in
- Observe the smtp email address has been removed

## Context
When our add-in is invoked by the user we call `Office.context.mailbox.item.to.getAsync(callback)` to get the current state of `to` recipients.  Depending on how the user has input the smtp mail address this information will be removed.

## Your Environment
* Platform [PC desktop, Mac, iOS, Office Online]: PC Desktop
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: 1908 (build 16.0.11929.20288)
* Operating System: Windows 10 (1903)
* Browser (if using Office Online): ______

**Comment by exextoc** [comment]
I'm unable to repro your issue, could you share a minimal example add-in?

**Comment by rgmills** [comment]
@exextoc Here's a repro.  As you found, it wasn't simply just calling `getAsync` that caused the issue.  The actual issue is calling `setAsync` on another recipient collection causes other recipient collections to resolve.  The same steps from above for reproducing apply.

Two observations:
- Outlook for Web does not exhibit this behavior
- If I switch to `addAsync` the issue does not occur.

```html<!DOCTYPE html>
<html>
	<head>
		<script type="text/javascript" src="https://appsforoffice.microsoft.com/lib/1.1/hosted/office.js"></script>
		<script type="text/javascript">
			function clicked() {
				var bccContacts = [];
				bccContacts.push({
					emailAddress: 'test@test.com',
					displayName: 'Test Test'
				});
				Office.context.mailbox.item.bcc.setAsync(bccContacts);
			}
		</script>
	</head>
	<body>
		<button onclick="clicked()">
			Run
		</button>
	</body>
</html>
```

**Comment by exextoc** [comment]
Thanks for reporting this issue regarding recipients. It has been put on our backlog. We unfortunately have no timelines to share at this point

**Comment by exextoc** [comment]
We have fixed the bug and the fix should be available in build 16.0.13330.10000 or later. Depending on the release channel a customer is on, getting an updated build can take anywhere from 4-8 weeks.
